File: Praktikum-2/Tutor/A/Rekening.cpp
#include "Rekening.h"

Rekening::Rekening(double x) {
	if (x < 0) {
		saldo = 0;
	} else {
		saldo = x;
	}
}

void Rekening::setSaldo(double x) {
	saldo = x;
}

double Rekening::getSaldo() const {
	return saldo;
}

void Rekening::simpanUang(double x) {
	saldo += x;
}

bool Rekening::tarikUang(double x) {
	if (x <= saldo) {
		saldo -= x;
		return 1;
	} else {
		return 0;
	}
}

File: Praktikum-4/Tutor/C/Rekening.java
abstract class Rekening {
	protected String nama;
	protected double saldo;
	protected double sukuBunga;

	public Rekening(String nama, double saldo, double sukuBunga) {
		this.nama = nama;
		if (saldo < 0) {
			this.saldo = 0;
		} else {
			this.saldo = saldo;
		}
		this.sukuBunga = sukuBunga;
	}

	abstract public void setor(double x);
	abstract public void tarik(double x);
	abstract public void update();

	public double hitungBiaya() {
		if (0.1 * this.saldo > 10) {
			return 10;
		} else {
			return 0.1 * this.saldo;
		}
	}

	public String getNama() {
		return this.nama;
	}

	public double getSaldo() {
		return this.saldo;
	}

	public double getSukuBunga() {
		return this.sukuBunga;
	}
}